core/lib/themes.js

```diff
@@ -8,10 +7,37 @@ import * as themeMixins from 'theme/mixins'
 import * as coreFilters from 'core/filters'
 import * as themeFilters from 'theme/filters'
 
+/**
+ *  Converts an Array to an Object
+ */
+function convertToObject (array) {
+  const obj = {}
+  array.forEach((element, index, array) => {
+    obj[element[0]] = element[1]
+  })
+  return obj
+}
+
 /**
  * Return messages load from both: core/resource/i18n and theme/resource/i18n
  */
 export function translations () {
+  const coreMessages = {
+    'en-US':
+      convertToObject(require('dsv-loader?rows!core/resource/i18n/en-US.csv')),
+    'de-DE':
```

Comment:
I’m just wondering about the optimization: can we add the conversion from CSV to JSON as a build step (you may just extend package.js ‘build’ and ‘dev’ scripts to another command); the command itself should be placed in core/scripts and used
To generate the /dist/i18n.js

I’m not sure about processing the /theme/*/resource/i18n.json -
Is your solution compliant with this feature of theme related translations?